scenario.py

```diff
@@ -18,6 +18,8 @@
 
 
 class Scenario:
+    fit_count = 0
```

Comment:
I am not sure this should be outside of the `__init__`. Is there a particular reason @Thomas-Galtier?

Reply:
I used a class variable to do the count, because i could not use the global keyword in not_twice_characteristic function (as it is a function defined inside an other function, python was lost with whether the variable are local or global).

Reply:
OK I see. I don't have the best practices about variable scopes in mind, but this looks a bit "hacky" to me. @natct10, @RomainGoussault would have a recommendation on how to do that properly?